fix: memberships sync

adekbadek opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Fixes the memberships sync (#40). If a membership is created, it means that it already exists on a remote site (it was synced to the site running the sync). 

### How to test the changes in this Pull Request:

This can't actually be tested until memberships backfill (https://github.com/Automattic/newspack-network/pull/62) is in place.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

adekbadek committed:
fix: memberships sync

leogermani reviewed (commented):
The goal is to prevent the metadata from being added when running the action a second time?

I think we need to add some safeguards there, because $user_membership can be a WP_Error object

Maybe add a `$is_creating` variable and only update the meta then?

adekbadek committed:
feat: update safeguards

adekbadek commented:
> The goal is to prevent the metadata from being added when running the action a second time?

Not exactly, the goal is to **only** add the metadata if the membership is a "cloned" one. 

> I think we need to add some safeguards there, because $user_membership can be a WP_Error object

The return type is `\WC_Memberships_User_Membership|null`, so I've added explicit `null` handling in 5b093a0cd2071878ff651bea09b36a943f0ea5e4.

Also, updated the safeguard against inactive plugin – as we've learned recently (https://github.com/Automattic/newspack-newsletters/pull/1437#discussion_r1494491359) it's best to directly check for the functions used.
